### ISSUE TITLE
When max transfers set, limit sync backlog to the max-transfer amount

### ISSUE THREAD

**Issue by beyondmeat** [rclone/rclone#3974]
#### What is your current rclone version (output from `rclone version`)?
1.51.0

#### What problem are you are trying to solve?
When using sync with --max-transfer, it would be nice to automatically limit the sync backlog to stop after reaching the limit set by --max-transfer. 

#### How do you think rclone should be changed to solve that?
There is no point in continuing to build the backlog beyond that point, so it should stop getting more files to upload once it gathered enough to fulfill the max-transfer amount.

**Comment by goku321** [comment]
Would like to give it a try.

**Comment by ncw** [comment]
This should be reasonably easy - I'd love to have your help @goku321 

Let me know if you want any guidance

**Comment by goku321** [comment]
Would below logic be sufficient here? https://github.com/rclone/rclone/blob/bcbe393af32bc6c6a388ce5e203e559e18f32aa3/fs/sync/sync.go#L129
```
backlog := ci.MaxBacklog
maxTransfer := int(ci.MaxTransfer)
if maxTransfer != -1 && backlog > maxTransfer {
    backlog = maxTransfer
}
```
Also, should I make `MaxBacklog` an `int64` ?

**Comment by ncw** [comment]
> Would below logic be sufficient here?
> 
> https://github.com/rclone/rclone/blob/bcbe393af32bc6c6a388ce5e203e559e18f32aa3/fs/sync/sync.go#L129
> 
> ```
> backlog := ci.MaxBacklog
> maxTransfer := int(ci.MaxTransfer)
> if maxTransfer != -1 && backlog > maxTransfer {
>     backlog = maxTransfer
> }
> ```

I don't think that quite works - `ci.MaxTranfser` is a size not a number of entries.

What is needed is that we stop putting things into the backlog when MaxTransfer is reached.

Effectively we need to stop the sync when sum(transferred)+sum(backlog) is breached.
